server/src/test/java/com/adobe/testing/s3mock/its/ListObjectIT.java

```diff
@@ -0,0 +1,194 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.adobe.testing.s3mock.its;
+
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+
+import com.adobe.testing.s3mock.S3MockApplication;
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.SdkClientException;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.ListObjectsRequest;
+import com.amazonaws.services.s3.model.ObjectListing;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.logging.log4j.util.Strings;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RunWith(Parameterized.class)
+public class ListObjectIT {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ListObjectIT.class);
+
+  private static final String BUCKET_NAME = "list-objects-test";
+
+  private static final String[] ALL_OBJECTS = 
+      arr("a", "b", "b/1", "b/1/1", "b/1/2", "b/2", "c/1", "c/1/1");
+
+  private static final boolean RUN_AGAINST_AWS = false;
+  
+  private static <T> T[] arr(@SuppressWarnings("unchecked") final T... a) {
+    return a;
+  }
+
+  /**
+   * Parameter fatcory.
+   * 
+   * @return
+   */
+  @Parameters(name = "{index}: prefix={0}, delimiter={1}")
+  public static Iterable<Object[]> data() {
+    return Arrays.<Object[]>asList(//
+        arr(null, null, ALL_OBJECTS, arr()), //
+        arr("", null, ALL_OBJECTS, arr()), //
+        arr(null, "", ALL_OBJECTS, arr()), //
+        arr("/", null, arr(), arr()), //
+        arr("b", null, arr("b", "b/1", "b/1/1", "b/1/2", "b/2"), arr()), //
+        arr("b/", null, arr("b/1", "b/1/1", "b/1/2", "b/2"), arr()), //
+        arr("b", "/", arr("b"), arr("b/")), //
+        arr("b/", "/", arr("b/1", "b/2"), arr("b/1/")), //
+        arr("b/1", "/", arr("b/1"), arr("b/1/")), //
+        arr("b/1/", "/", arr("b/1/1", "b/1/2"), arr()), //
+        arr("c", "/", arr(), arr("c/")), //
+        arr("c/", "/", arr("c/1"), arr("c/1/")) //
+    );
+  }
+
+  private static final List<String> param = new ArrayList<>();

```

Comment:
This looks like debug stuff... can we remove that?

Reply:
This list was indeed used during development to collect the answers by AWS in order to build the expectations. I kept it in, because it might come in handy if someone wants to add more cases, but I can simple keep my development version around and get rid of it here.